big_tests/tests/rest_SUITE.erl

```diff
@@ -185,17 +241,22 @@ sessions_are_listed(_) ->
     true = is_list(Sessions).
 
 session_can_be_kicked(Config) ->
-    escalus:story(Config, [{alice, 1}], fun(Alice) ->
+    escalus:fresh_story(Config, [{alice, 1}], fun(Alice) ->
         % Alice is connected
-        Domain = domain(),
+        AliceJid = jid:nameprep(escalus_client:full_jid(Alice)),
+        AliceSessionPath = <<"/sessions/", (escalus_client:server(Alice))/binary,
+                             "/", (escalus_client:username(Alice))/binary,
+                             "/", (escalus_client:resource(Alice))/binary>>,
         {?OK, Sessions1} = gett(admin, "/sessions/localhost"),
-        assert_inlist(<<"alice@", Domain/binary, "/res1">>, Sessions1),
+        assert_inlist(AliceJid, Sessions1),
         % kick alice
-        {?NOCONTENT, _} = delete(admin, "/sessions/localhost/alice/res1"),
+        {?NOCONTENT, _} = delete(admin, AliceSessionPath),
         escalus:wait_for_stanza(Alice),
         true = escalus_connection:wait_for_close(Alice, timer:seconds(1)),
         {?OK, Sessions2} = gett(admin, "/sessions/localhost"),
-        assert_notinlist(<<"alice@", Domain/binary, "/res1">>, Sessions2)
+        assert_notinlist(AliceJid, Sessions2),
+        {?FORBIDDEN, <<"no active session">>} = delete(admin, AliceSessionPath),
```

Comment:
why not 404?

Reply:
true - since jids are now in path, we can treat it as a resource which may not be there.